lib/channel-postmessage/src/index.ts

```diff
@@ -89,25 +105,56 @@ export class PostmsgTransport {
     });
   }
 
-  private getWindow(): Window {
+  private getFrames(target?: string): Window[] {
     if (this.config.page === 'manager') {
-      // FIXME this is a really bad idea! use a better way to do this.
-      // This finds the storybook preview iframe to send messages to.
-      const iframe = document.getElementById('storybook-preview-iframe');
-      if (!iframe) {
-        return null;
-      }
-      return iframe.contentWindow;
+      const nodes: HTMLIFrameElement[] = [
+        ...document.querySelectorAll('#storybook-preview-wrapper iframe'),
+      ];
+
+      const list = nodes
+        .filter(e => {
+          try {
+            return !!e.contentWindow && e.dataset.isStorybook !== undefined && e.id === target;
```

Comment:
Does this code work if `target` is undefined? Can you write a comment explaining what it is doing?

Reply:
if target is undefined, it will get the currentFrame:

> `return list.length ? list : this.getCurrentFrame();`